code/game/gamemodes/traitor/double_agents.dm

```diff
@@ -18,29 +22,179 @@
 	var/list/target_list = list()
 	var/list/late_joining_list = list()
 
+
 /datum/game_mode/traitor/internal_affairs/post_setup()
 	var/i = 0
 	for(var/datum/mind/traitor in traitors)
 		i++
 		if(i + 1 > traitors.len)
 			i = 0
-		target_list[traitor] = traitors[i + 1]
+		target_list[traitor] = traitors[i+1]	
 	..()
 
+
+/datum/action/agent_pinpointer

```

Comment:
This could probably just be a /datum/status_effect instead, this is kinda a weird solution; status effects already have processing and can do everything this can.
Also, they don't appear on the action bar, which is a weird place for the pinpointer to show up in the first place.